markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java

```diff
@@ -567,18 +567,20 @@ public static String getMarkdownLink(@NonNull String text, @NonNull String url)
      * @return Plain text string
      */
     @NonNull
-    // CS304 issue link: https://github.com/stefan-niedermann/nextcloud-notes/issues/1212
     public static String removeMarkdown(@Nullable String s) {
         if (s == null)
             return "";
-        // TODO maybe we can utilize the markwon renderer?
         // Create HTML string from Markup
         String html = renderer.render(parser.parse(replaceCheckboxesWithEmojis(s)));
-        html = html.replaceAll("\n","");
-        // Create Spanned from HTML, with special handling for ordered list items
-        Spanned spanned = Html.fromHtml(html,Html.FROM_HTML_MODE_COMPACT);
-        String ans = spanned.toString();
-        ans = ans.substring(0,ans.length()-1);
-        return ans;
+        html = html.replaceAll("\n", "");
+        // Convert Spanned from HTML. Using HtmlCompat to convert has some problems, which is that sometimes the \n in the text would double.
+        Spanned spanned = HtmlCompat.fromHtml(html, 0);
```

Comment:
Have you tried to pass `HtmlCompat.FROM_HTML_MODE_COMPACT` instead of `0`? Maybe this solves the problem with the doubled `\n`?

Reply:
After trying HtmlCompat.FROM_HTML_MODE_COMPACT instead of 0, the problem of dobled \n is solved. Thanks very much!
Is there any other questions I need to deal with? 😃